src/averageAges.js

```diff
@@ -37,7 +48,21 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+  const mothers = people.map(x => x.mother);
+  const mothersWithChild = people.filter(x => mothers.includes(x.name));
+  const ageOfDeath = women.map(person => person.died - person.born);
+  const ageOfDeathMom = mothersWithChild.map(person =>
+    person.died - person.born
+  );
+
+  const result = !withChildren ? (
+    ageOfDeath.reduce((sum, age) => sum + age, 0) / women.length
+  ) : (
+    ageOfDeathMom.reduce((sum, age) => sum + age, 0) / mothersWithChild.length
+  );
+
+  return result;
```

Comment:
I think your solution looks better now, but it still overcomplicated, so here's an alternative solution, maybe it will be useful if you go step by step and analyze it for the future ;)

```
  let women = people.filter(person => person.sex === 'f');

  if (withChildren) {
    women = women.filter(woman =>
      people.some(person => person.mother === woman.name));
  }

  return women.reduce((sum, woman) =>
    sum + (woman.died - woman.born), 0
  ) / women.length;

```